presto-main/src/test/java/io/prestosql/server/TestQueryResource.java

```diff
@@ -111,6 +144,49 @@ public void testGetQueryInfoExecutionFailure()
         assertEquals(info.getFailureInfo().getErrorCode(), DIVISION_BY_ZERO.toErrorCode());
     }
 
+    @Test
+    public void testCancel()
+    {
+        String queryId = startQuery("SELECT * FROM tpch.sf100.lineitem");
+
+        server.getAccessControl().deny(privilege("query", KILL_QUERY));
+        try {
+            assertEquals(cancelQueryInfo(queryId), 403);
+        }
+        finally {
+            server.getAccessControl().reset();
+        }
+
+        assertEquals(cancelQueryInfo(queryId), 204);
+    }
+
+    @Test
+    public void testKilled()
+    {
+        testKilled("killed");
+    }
+
+    @Test
+    public void testPreempted()
+    {
+        testKilled("preempted");
+    }
+
+    private void testKilled(String preempted)
+    {
+        String queryId = startQuery("SELECT * FROM tpch.sf100.lineitem");
+
+        server.getAccessControl().deny(privilege("query", KILL_QUERY));
+        try {
+            assertEquals(killQueryInfo(queryId, preempted), 403);
+        }
+        finally {
+            server.getAccessControl().reset();
+        }
+
+        assertEquals(killQueryInfo(queryId, preempted), 202);
```

Comment:
can you run query `system.runtime.queries` to see that query was killed and possibly check the reason why query was killed